Set suggested display precision in modbus integration

slickm0nty opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
Updates the sensor entity to define suggested display precision the same as data ingestion


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.

  AI tools are welcome, but contributors are responsible for *fully*
  understanding the code before submitting a PR.
-->

- [x] I understand the code I am submitting and can explain how it works.
- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.
- [x] Any generated code has been carefully reviewed for correctness and compliance with project standards.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

slickm0nty committed:
Add suggested ui attributes to modbus integration

slickm0nty committed:
Merge branch 'dev' into modbus_suggest_attr

slickm0nty commented:
> But why? I am missing what the component doesn't do what it should do with this change

The UI defaults to INT for precision and local temp for temp values. https://developers.home-assistant.io/blog/2025/05/26/sensor-default-display-precision/

Generally that wouldn't be an issue, but for my use case I am pulling data from multiple LifePO4 battery packs and the temps were auto-converting to Fahrenheit and cell voltages were rounding, making it more difficult to determine if I had the right register, offset, and data type, etc. 

Changing the data presentation can be done manually through the UI, but adjusting 30-40 sensors on each pack would not be ideal.

abmantis reviewed (commented):
I agree on setting `_attr_suggested_display_precision`, since integrations should set that when the fallback implementation is not enough.

For `_attr_suggested_unit_of_measurement` I am not 100% sure. Would users usually want the unit to be the native one from the device, or the default on on their HA configuration?

slickm0nty committed:
Merge branch 'dev' into modbus_suggest_attr

frenck commented:
Converted the PR to a draft.

@slickm0nty The CI seems to be failing, can you take a look? Additionally, there is an open question above by @abmantis.

../Frenck

slickm0nty committed:
Fix for failing tests and removal of default temp

slickm0nty committed:
Merge branch 'dev' into modbus_suggest_attr

slickm0nty commented:
> I agree on setting `_attr_suggested_display_precision`, since integrations should set that when the fallback implementation is not enough.
> 
> For `_attr_suggested_unit_of_measurement` I am not 100% sure. Would users usually want the unit to be the native one from the device, or the default on on their HA configuration?

Not setting unit of measurement seems like the right call. I've removed it and updated the pr. 

Also fixed the issue being raised in the CI pipeline

abmantis reviewed (approved):
Thanks @slickm0nty
